superset/db_engine_specs/base.py

```diff
@@ -397,6 +397,9 @@ class BaseEngineSpec:  # pylint: disable=too-many-public-methods
     # Can the catalog be changed on a per-query basis?
     supports_dynamic_catalog = False
 
+     # Use column alias instead of aggregate function in GROUP BY
+    use_column_alias_in_groupby = False
```

Comment:
I'm surprised that the SQLAlchemy dialect doesn't handle this. The [`visit_label`](https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/sql/compiler.py#L2479-L2526) method is likely relevant. Ideally we adopt the "shift left" mentality and have the underlying dialect (rather than Superset) handle this logic.